src/sentry/logging/handlers.py

```diff
@@ -45,9 +33,38 @@
 )
 
 
+def _get_JSONEncoder(skipkeys=False):
+    return json.JSONEncoder(
+        separators=(",", ":"),
+        ignore_nan=True,
+        skipkeys=skipkeys,
+        ensure_ascii=True,
+        check_circular=True,
+        allow_nan=True,
+        indent=None,
+        encoding="utf-8",
+        default=_json_fallback_handler,
+    )
+
+
+_JSONEncoder_skipkeys = _get_JSONEncoder(skipkeys=True)
+_JSONEncoder_no_skipkeys = _get_JSONEncoder(skipkeys=False)
+
+
 class JSONRenderer:
     def __call__(self, logger, name, event_dict):
-        return _default_encoder(event_dict)
+        # importing inside of the function to avoid circular imports
+        from django.conf import settings  # NOQA isort:skip # pylint:disable=import-outside-toplevel
+
+        try:
+            return _JSONEncoder_no_skipkeys.encode(event_dict)
+        except Exception:
+            logging.warning("Failed to serialize event", exc_info=True)
+            # in Production, we want to skip non-serializable keys, rather than raise an exception
+            if settings.DEBUG:
+                raise
+            else:
+                return _JSONEncoder_skipkeys.encode(event_dict)
```

Comment:
@hubertsentry is this what you had in mind to surface these errors?

Reply:
yeah, it looks good to me